subsys/bluetooth/host/audio/mpl_internal.h

```diff
@@ -84,6 +84,12 @@ struct mpl_mediaplayer_t {
 #endif /* CONFIG_BT_OTS || CONFIG_BT_OTC */
 	uint8_t              content_ctrl_id;
 	struct media_proxy_pl_calls calls;
+
+	bool                        next_track_set; /* If next track explicitly set */
+	struct {
+		struct mpl_track_t  *track; /* The track explicitly set as next track */
+		struct mpl_group_t  *group; /* The group of the set track */
```

Comment:
Is it only the track that can be explicitly set, or also the group?

Reply:
For "next track", it is the track that is set.  The group of the track is stored to avoid having to do a "find" again later.

Reply:
I see :) I guess the `group` of the `track` isn't stored in the `mpl_track_t` struct right, since we need this here?

Reply:
Exactly, that is how it is.
As I have commented above, the data structure could be changed so that each track has a pointer to it's group (which would be fun to do, ad would also help with the "set next track"), but I do not think that is worth the effort now.